File: solution/210826/주식가격.py
def solution(prices):
    answer = [len(prices)-i-1 for i in range(len(prices))]
    stack = []
    for i, p in enumerate(prices):
        while stack:
            print(stack)
            if stack[-1][0] > p:
                pop = stack.pop()
                answer[pop[1]] = i-pop[1]
            else:
                break
        stack.append([p, i])

    print(answer)
    return answer

File: solution/211002/주식가격.java
import java.util.*;
import java.util.stream.Collectors;
class Solution {

    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> stack = new Stack<>();
        stack.addAll(Arrays.stream(prices).boxed().collect(Collectors.toList()));

        Deque<Integer> deque = new ArrayDeque<>();

        for (int i = prices.length-1 ; i>=0; i--){
            int count =-1;
            Integer pop = stack.pop();
            deque.addFirst(pop);
            for (Integer price : deque) {
                count++;
                if (prices[i] > price) break;
            }
            answer[i] = count;
        }
        
        return answer;
    }

}
